### PR TITLE
Adapt binaries to work under relative paths

### PR DESCRIPTION
**Pull request by alromeros**
|Related issue|
|---|
|#7430|

## Description

Hi team,

This pull request aims to include several changes to adapt Wazuh's binaries to work under relative paths, thus making the compiling process independent of any predefined installation path. Several changes have been introduced to achieve this:

- Each binary has first to move its working directory with `chdir()` to the one in which it's installed.
- From now on, binaries use relative paths to use Wazuh-related resources.
- When outputting path-related data, It's favorable to print the absolute path as usual, generally using `abspath()`.

## Configuration options

All manual configuration including path usage is meant to work in the same way as usual.

## Logs/Alerts example

Despite all modifications, the standard installation should provide a successful log:

````
...

Processing dependencies for api==4.2.0
Finished processing dependencies for api==4.2.0
cd ../tools/mitre && /var/ossec/framework/python/bin/python3 mitredb.py -d /var/ossec/var/db/mitre.db
Generating self-signed certificate for wazuh-authd...

Starting Wazuh...

 - Configuration finished properly.

 - To start Wazuh:
      /var/ossec/bin/wazuh-control start

 - To stop Wazuh:
      /var/ossec/bin/wazuh-control stop

 - The configuration can be viewed or modified at /var/ossec/etc/ossec.conf


   Thanks for using Wazuh.
   Please don't hesitate to contact us if you need help or find
   any bugs.

   Use our public Mailing List at:
          https://groups.google.com/forum/#!forum/wazuh

   More information can be found at:
          - http://www.wazuh.com

    ---  Press ENTER to finish (maybe more information below). ---


 - In order to connect agent and server, you need to add each agent to the server.

   More information at: 
   https://documentation.wazuh.com/
````

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
  - [x] Windows
  - [x] MAC OS X
- [x] Source installation
- [x] Source upgrade

- Memory tests for Linux
  - [x] Scan-build report
- Memory tests for Windows
  - [ ] Scan-build report
- Memory tests for macOS
  - [x] Scan-build report

- Check Unit Tests:
  - [x] Server
  - [x] Agent
  - [x] Winagent

### CODE REVIEW

File: src/remoted/shared_download.c [hunk 1]
```diff
@@ -532,7 +532,7 @@ void w_create_group(char *group){
 
     char group_path[PATH_MAX] = {0};
 
-    if(snprintf(group_path,PATH_MAX,isChroot() ? "/etc/shared/%s" : BUILDDIR(HOMEDIR,"/etc/shared/%s"), group) >= PATH_MAX){
+    if(snprintf(group_path,PATH_MAX, "/etc/shared/%s", group) >= PATH_MAX){
```
Comment (chemamartinez) [inline src/remoted/shared_download.c:535]: Make this path relative and use the define `SHAREDCFG_DIR`:
```
snprintf(group_path,PATH_MAX, "%s/%s", SHAREDCFG_DIR, group)
```

**Review by chemamartinez — CHANGES_REQUESTED** [review]

File: src/unit_tests/syscheckd/test_syscheck_config.c [hunk 2]
```diff
@@ -44,9 +44,6 @@ void test_Read_Syscheck_Config_success(void **state)
     expect_any_always(__wrap__mdebug1, formatted_msg);
     expect_any_always(__wrap__mwarn, formatted_msg);
 
-#ifdef TEST_AGENT
-    will_return_always(__wrap_isChroot, 1);
```
Comment (chemamartinez) [inline src/unit_tests/syscheckd/test_syscheck_config.c:48]: Same

File: src/unit_tests/wazuh_db/test_wdb_agent.c [hunk 3]
```diff
@@ -144,7 +144,6 @@ void test_wdb_create_agent_db_error_opening_dest_profile(void **state)
     expect_string(__wrap_fopen, mode, "r");
     will_return(__wrap_fopen, 1);
     // Opening destination database file
-    will_return(__wrap_isChroot, 0);
```
Comment (chemamartinez) [inline src/unit_tests/wazuh_db/test_wdb_agent.c:147]: Same

File: src/win32/agent_auth.c [hunk 4]
```diff
@@ -313,10 +313,12 @@ void InstallAuthKeys(char *msg)
...
 
-        fp = fopen(KEYSFILE_PATH, "w");
+        fp = fopen(KEYS_FILE, "w");
 
-        if (!fp)
-            merror_exit("Unable to open key file: %s", KEYSFILE_PATH);
+        if (!fp) {
+            merror_exit("Unable to open key file: %s", KEYS_FILE);
+        }
+
```
Comment (chemamartinez) [inline src/win32/agent_auth.c:321]: Please remove this extra line

**Review by chemamartinez — CHANGES_REQUESTED** [review]

File: src/wazuh_modules/wmodules.h [hunk 5]
```diff
@@ -17,8 +17,8 @@
 #include "config/config.h"
 #include "wmodules_def.h"
 
-#define WM_DEFAULT_DIR  BUILDDIR(HOMEDIR,"/wodles")       // Default modules directory.
-#define WM_STATE_DIR    BUILDDIR(HOMEDIR,"/var/wodles")   // Default directory for states.
+#define WM_DEFAULT_DIR  "wodles"                   // Default modules directory.
```
Comment (chemamartinez) [inline src/wazuh_modules/wmodules.h:20]: This `WM_DEFAULT_DIR` is not used in the project. Remove it.

File: src/wazuh_modules/syscollector/syscollector_common.c [hunk 6]
```diff
@@ -0,0 +1,322 @@
+/*
```
Comment (chemamartinez) [inline src/wazuh_modules/syscollector/syscollector_common.c:1]: Please remove this file. It was deleted in another development.

**Review by chemamartinez — CHANGES_REQUESTED** [review]

File: src/wazuh_modules/wmodules.h (hunk 5 above)
Comment (danimegar) [reply src/wazuh_modules/wmodules.h]: It was removed in this commit fcff159439fdcee38c8870c942baacb153c0954f

File: src/wazuh_modules/syscollector/syscollector_common.c (hunk 6 above)
Comment (danimegar) [reply src/wazuh_modules/syscollector/syscollector_common.c]: It was removed in this commit fcff159439fdcee38c8870c942baacb153c0954f

**Review by chemamartinez — APPROVED** [review]

**Pull request merged**